include/LightGBM/c_api.h

```diff
@@ -180,8 +189,8 @@ DllExport int LGBM_DatasetSetField(DatesetHandle handle,
 * \param field_name field name
 * \param out_len used to set result length
 * \param out_ptr pointer to the result
-* \param out_type  float_32:0, int32_t:1
-* \return 0 when success, -1 when failure happens
+* \param out_type  float32 or int32
```

Comment:
Alright, sounds good. Let's add a note to the C API function to indicate that the function won't work when called on a "label" with `type == C_API_DTYPE_INT32`.